src/background.js

```diff
@@ -65,12 +67,18 @@ class Crawler {
     }
 
     const response = await fetch(url);
+    let responseText;
     if (response.ok) {
-      const responseText = await response.text();
+      responseText = await response.text();
```

Comment:
You can just use a ternary there: `const responseText = response.ok ? await.response.text() : null;`

Then just return if responseText is not falsy `if (!responseText) return;`